satpy/utils.py

```diff
@@ -43,16 +44,83 @@ def ensure_dir(filename):
         os.makedirs(directory)
 
 
-def debug_on():
-    """Turn debugging logging on."""
+def debug_on(dep_warnings=True):
+    """Turn debugging logging on.
+
+    Sets up a StreamHandler to to `sys.stderr` at debug level for all
+    loggers, such that all debug messages (and log messages with higher
+    severity) are logged to the standard error stream.
+
+    By default, since Satpy 0.26, this also enables the global visibility
+    of deprecation warnings.  This can be suppressed by passing a false
+    value.
+
+    Args:
+        dep_warnings (Optional[bool]): Switch on deprecation warnings.
+            Defaults to True.
+
+    Returns:
+        None
+    """
     logging_on(logging.DEBUG)
+    if dep_warnings:
+        dep_warnings_on()
+
+
+def debug_off():
+    """Turn debugging logging off.
+
+    This disables both debugging logging and the global visibility of
+    deprecation warnings.
+    """
+    logging_off()
+    dep_warnings_off()
+
+
+@contextlib.contextmanager
+def debug(dep_warnings=True):
+    """Context manager to temporarily set debugging on.
+
+    Example::
+
+        >>> with satpy.utils.debug():
+        ...     code_here()
+
+    Args:
+        dep_warnings (Optional[bool]): Switch on deprecation warnings.
+            Defaults to True.
+    """
+    debug_on()
+    yield
+    debug_off()
 
 
 def trace_on():
     """Turn trace logging on."""
     logging_on(TRACE_LEVEL)
 
 
+class _WarningManager:
+    """Class to handle switching warnings on and off."""
+
+    filt = None
+
+
+_warning_manager = _WarningManager()
+
+
+def dep_warnings_on():
```

Comment:
`dep` can be missleading (I was thinking dependency, as in warnings from other libraries), maybe just rename to `deprecation_warnings_on` for clarity?